index.js

```diff
@@ -43,7 +45,7 @@ class IsConnected extends EventEmitter {
          */
     _dns() {
             if (os.platform() === 'linux') {
-                setInterval(function() {
+                setTimeout(function() {
```

Comment:
I am initiating another setTimout as soon as the ping or dns lookup completes. So it does run continuously. The problem with using setInterval is that if the ping/dns takes longer than the interval the calls begin to queue up. Then when network connectivity is restored they all execute immediately and you get a lot of connection events all at once.